rasa/nlu/featurizers/dense_featurizer/lm_featurizer.py

```diff
@@ -42,58 +45,102 @@
 logger = logging.getLogger(__name__)
 
 
-class LanguageModelFeaturizer(DenseFeaturizer):
-    """Featurizer using transformer-based language models.
+class LanguageModelFeaturizerGraphComponent(DenseFeaturizer2, GraphComponent):
+    """A featurizer that uses transformer-based language models.
 
-    The transformers(https://github.com/huggingface/transformers) library
-    is used to load pre-trained language models like BERT, GPT-2, etc.
-    The component also tokenizes and featurizes dense featurizable attributes of
+    This component loads a pre-trained language model
+    from the Transformers library (https://github.com/huggingface/transformers)
+    including BERT, GPT, GPT-2, xlnet, distilbert, and roberta.
+    It also tokenizes and featurizes the featurizable dense attributes of
     each message.
     """
 
-    defaults = {
-        # name of the language model to load.
-        "model_name": "bert",
-        # Pre-Trained weights to be loaded(string)
-        "model_weights": None,
-        # an optional path to a specific directory to download
-        # and cache the pre-trained model weights.
-        "cache_dir": None,
-    }
-
-    @classmethod
-    def required_components(cls) -> List[Type[Component]]:
-        """Packages needed to be installed."""
-        return [Tokenizer]
-
     def __init__(
         self,
-        component_config: Optional[Dict[Text, Any]] = None,
-        skip_model_load: bool = False,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
     ) -> None:
-        """Initializes LanguageModelFeaturizer with the specified model.
-
-        Args:
-            component_config: Configuration for the component.
-            skip_model_load: Skip loading the model for pytests.
-        """
-        super(LanguageModelFeaturizer, self).__init__(component_config)
+        """Initializes LanguageModelFeaturizer with the model
+        specified in the config."""
+        super(LanguageModelFeaturizerGraphComponent, self).__init__(
+            execution_context.node_name, config
+        )
+        self._config = {**self.get_default_config(), **config}
+        self._model_storage = model_storage
+        self._resource = resource
+        self._execution_context = execution_context
```

Comment:
these are actually not used anywhere, are they?
```suggestion
```

Reply:
Oh yeah. I must have copied it from one of the other classes. Oops!